crates/metadata/src/layout/mod.rs

```diff
@@ -146,25 +149,37 @@ impl IntoPortable for RootLayout {
         RootLayout {
             root_key: self.root_key,
             layout: Box::new(self.layout.into_portable(registry)),
+            ty: registry.register_type(&self.ty),
         }
     }
 }
 
 impl<F> RootLayout<F>
 where
     F: Form,
+    F::Type: From<MetaType>,
```

Comment:
With this constraint the impl is no longer generic over `Form`. If we wanted to use the rest of the methods e.g. on a decoded `RootLayout<PortableForm>` then the methods will not be available.

The way we separate the `MetaForm` constructors in other places is to have a separate non generic `MetaForm` impl e.g.

```rust
impl RootLayout<MetaForm> {
    /// Creates a new root layout.
    pub fn new<Root, L>(root_key: LayoutKey, layout: L) -> Self
    where
        Root: TypeInfo + 'static,
        L: Into<Layout<MetaForm>>,
    {
        Self {
            root_key,
            layout: Box::new(layout.into()),
            ty: meta_type::<Root>(),
        }
    }

    /// Creates a new root layout with empty root type.
    pub fn new_empty<L>(root_key: LayoutKey, layout: L) -> Self
    where
        L: Into<Layout<MetaForm>>,
    {
        Self::new::<(), L>(root_key, layout)
    }
}
```

Reply:
You are right, my bad=D